File: main/dynamicProgramming/fibonnaci.py
def fibonnaci(n):
  cache = {}
  return _fibonnaci(n, cache)

def _fibonnaci(n, cache):
  if n == 0 or n == 1:
    return n
  
  if n in cache:
    return cache[n]
  
  cache[n] = _fibonnaci(n - 1, cache) + _fibonnaci(n - 2, cache)
  
  return cache[n]

File: main/dynamic-programming/fibonnaci.ts
function fibonnaci(n: number, cache: number[] = []) {
    if (n === 0 || n === 1) return n;

    if (n in cache) return cache[n];

    cache[n] = fibonnaci(n - 1, cache) + fibonnaci(n - 2, cache);

    return cache[n];
}
